src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -2536,10 +2550,22 @@ class CreateViewStatementSegment(BaseSegment):
 
 @ansi_dialect.segment()
 class DropStatementSegment(BaseSegment):
-    """A `DROP` statement."""
+    """A `DROP` statement.
+
+    R7L208:
+        - Adding Function into this drop statement conflicts with
+          DropFunctionStatementSegment in Postgres dialect causing
+          an unparsable section resulting in failure of
+          test/fixtures/dialects/postgres/postgres_drop_function.sql
+
+        - Adding TemporaryGrammar into DropStatementSegment throws
+          an error since TEMPORARY and TEMP are not keywords in tSQL
+          dialect.
+    """
```

Comment:
tbh it's pretty inconsistent that we have some drop statements combined into a single `DropStatementSegment` and others like `DropFunctionStatementSegment` and `DropDatabaseSegment` (the latter which being introduced in this PR).

For historical context, these DROP statements were likely initially combined so we could quickly get more of the basic grammar parsing. However, as we start to build out the statements with more dialect specific additions going forwards we should look to split them out into separate segments because:
a) We don't end up with any issues with some grammars splitting them and others not.
b) As the grammar gets more complicated it's easier to understand when a segment only covers one statement.
c) Matches documentation better, usually documentation splits these statements into their own pages.

Reply:
So with that being said could you split `DropStatementSegment` into
- [ ] `DropTableStatementSegment`
- [ ] `DropViewStatementSegment`
- [ ] `DropUserStatementSegment`

And also check where `DropStatementSegment` is being used in other dialects and split them out similarly.

Reply:
(Also don't think we need the note here, you've already overridden this grammar in spark3 so we don't need these extra things added to ANSI in this PR)

Reply:
Thought the note would be helpful for future attempts of extending this :) Since the decision is to split it out, then don't think it's needed.

I'm working on new commits for splitting it out

Reply:
awesome thanks 😄